File: src/main/java/com/likelionknu/applyserver/auth/data/repository/UserRepository.java
package com.likelionknu.applyserver.auth.data.repository;

import com.likelionknu.applyserver.auth.data.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    User findByEmail(String email);
    Optional<User> findOptionalByEmail(String email);
}

File: src/main/kotlin/auth/repository/UserRepository.kt
package com.likelionknu.applyserver.auth.repository


import com.likelionknu.applyserver.auth.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): User
    fun findOptionalByEmail(email: String): Optional<User>
}